### PR TITLE
WebUI: Fix changing category on HBAC/Sudo/etc Rule pages

### PR DESCRIPTION
**Pull request by serg-cymbaluk**
No object can be added to a rule when object category is 'all'.
So while editing rule there is needed to save actual category value before adding related objects.

Ticket: https://pagure.io/freeipa/issue/7961

### CODE REVIEW

File: ipatests/test_webui/test_selinuxusermap.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 HBAC_DEL_ERR = ('{} cannot be deleted because SELinux User Map {} requires '
                 'it')
 HBAC_MEMBER_ERR = 'HBAC rule and local members cannot both be set'
+BATCH_ERR = 'Some operations failed.'
```
Comment (netoarmando) [inline ipatests/test_webui/test_selinuxusermap.py]: Every other message doesn't end with a period.

**Comment by serg-cymbaluk** [comment]
@netoarmando 

> You have updated one existing test case; however, I think a test case is missing, for when 'all' is the current selection and the user switches that and tries to add an item to the list. Am I right?

Yes, we need to cover this issue, but not in the scope of the PR.

**Comment by netoarmando** [comment]
Ok! Looks good to me. I tested locally, the issue is gone and the new behaviour was clear from the user's perspective.
Please, remove the temp commit and I'll ack it.

**Comment by serg-cymbaluk** [comment]
@netoarmando 
Thanks for the review.
Temp commit is removed.

**Comment by serg-cymbaluk** [comment]
master:

* 755154318ac6b6a57eee2bdec76debd25e97fd45 WebUI: Fix changing category on HBAC/Sudo/etc Rule pages

**Pull request closed without merging**